PSReadLine/Render.cs

```diff
@@ -310,38 +311,61 @@ void MaybeEmphasize(int i, string currColor)
                     activeColor = string.Empty;
 
                     UpdateColorsIfNecessary(Options._continuationPromptColor);
-                    foreach (char c in Options.ContinuationPrompt)
-                    {
-                        _consoleBufferLines[currentLogicalLine].Append(c);
-                    }
+                    _consoleBufferLines[currentLogicalLine].Append(Options.ContinuationPrompt);
 
                     if (inSelectedRegion)
                     {
                         // Turn off inverse before end of line, turn on after continuation prompt
                         _consoleBufferLines[currentLogicalLine].Append(Options.SelectionColor);
                     }
                 }
+                else if (char.IsControl(charToRender))
+                {
+                    MaybeEmphasize(i, color);
+                    _consoleBufferLines[currentLogicalLine].Append('^');
+                    _consoleBufferLines[currentLogicalLine].Append((char)('@' + charToRender));
+                }
                 else
                 {
-                    if (char.IsControl(charToRender))
+                    MaybeEmphasize(i, color);
+                    _consoleBufferLines[currentLogicalLine].Append(charToRender);
+                }
+            }
+
+            if (inSelectedRegion)
+            {
+                _consoleBufferLines[currentLogicalLine].Append("\x1b[0m");
+            }
+
+            inSelectedRegion = false;
+            if (suggestion != null)
+            {
+                color = _options._predictionColor;
+                UpdateColorsIfNecessary(color);
+
+                foreach (char charToRender in suggestion)
+                {
+                    if (charToRender == '\n')
+                    {
+                        currentLogicalLine += 1;
+                        if (currentLogicalLine == _consoleBufferLines.Count)
+                        {
+                            _consoleBufferLines.Add(new StringBuilder(COMMON_WIDEST_CONSOLE_WIDTH));
+                        }
+                        _consoleBufferLines[currentLogicalLine].Append(Options.ContinuationPrompt);
+                    }
```

Comment:
The continuation prompt could change the color so you'll need to restore the prediction color.

Reply:
Fixed by removing the duplicate logic.